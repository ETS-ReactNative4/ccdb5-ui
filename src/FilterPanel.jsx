import './FilterPanel.less'
import CompanyName from './Filters/CompanyName'
import { connect } from 'react-redux'
import DateFilter from './Filters/DateFilter'
import FederalState from './Filters/FederalState'
import Issue from './Filters/Issue'
import Product from './Filters/Product'
import React from 'react'
import SimpleFilter from './Filters/SimpleFilter'
import SingleCheckbox from './Filters/SingleCheckbox'
import ZipCode from './Filters/ZipCode'

export class FilterPanel extends React.Component {
  render() {
    const descPublicResponse = "The company's optional public-facing " +
      "response to a consumer's complaint"
    const descConsumerConsent = 'Whether a consumer opted to publish their ' +
      'compaint narrative'

    return (
      <section className="filter-panel">
        <h3>Filter results by...</h3>
        <SingleCheckbox title="Only show complaints with narratives?"
                        fieldName="has_narrative" />
        <hr />
        <DateFilter fieldName="date_received"
                    title="Date CFPB Received the complaint" />
        <hr />
        <CompanyName />
        <hr />
        <Product />
        <hr />
        <Issue />
        <hr />
        <FederalState />
        <hr />
        <ZipCode />
        <hr />
        <SimpleFilter title="Did company provide a timely response?"
                      fieldName="timely"
        />
        <hr />
        <SimpleFilter title="Company Response"
                     desc="How the company responded to the complaint"
                     fieldName="company_response"
        />
        <hr />
        <SimpleFilter title="Company Public Response"
                     desc={descPublicResponse}
                     fieldName="company_public_response"
        />
        <hr />
        <DateFilter title="Date complaint sent to company"
                    fieldName="company_received"
        />
        <hr />
        <SimpleFilter title="Consumer Consent"
                     desc={descConsumerConsent}
                     fieldName="consumer_consent_provided"
        />
        <hr />
        <SimpleFilter title="How did the consumer submit the complaint to the CFPB?"
                     fieldName="submitted_via"
        />
      </section>
    )
  }
}

const mapStateToProps = state => ( {
  aggs: state.aggs
} )

export default connect( mapStateToProps )( FilterPanel )
