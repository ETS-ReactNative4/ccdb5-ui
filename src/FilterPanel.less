@import (less) "base.less";

.filter-panel {
    padding: @gutter-normal;
    border: 1px solid @gray-80;
    border-right: 0;  // Don't double the border next to the content

    background: @gray-5;
}