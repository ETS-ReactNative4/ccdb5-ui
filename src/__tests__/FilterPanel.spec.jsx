import React from 'react';
import FilterPanel from '../FilterPanel';
import renderer from 'react-test-renderer';

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = renderer.create(
      <FilterPanel />
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

