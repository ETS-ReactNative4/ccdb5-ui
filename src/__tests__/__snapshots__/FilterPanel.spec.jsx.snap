exports[`initial state renders without crashing 1`] = `
<section
  className="filter-panel">
  <h3>
    Filter results by...
  </h3>
  <section
    className="single-checkbox">
    <h5>
      Only show complaints with narratives?
    </h5>
    <div
      className="m-form-field m-form-field__checkbox">
      <input
        checked={false}
        className="a-checkbox"
        id="theCheckbox"
        onClick={[Function]}
        type="checkbox"
        value="has_narrative" />
      <label
        className="a-label"
        htmlFor="theCheckbox">
        Yes
      </label>
    </div>
  </section>
  <hr />
  <section
    className="o-expandable aggregation date-filter">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        Date CFPB Received the complaint
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
          onClick={[Function]}>
          Hide
          <span
            className="cf-icon cf-icon-minus-round" />
        </button>
      </span>
    </div>
    <p />
    <div>
      <ul
        className="date-inputs">
        <li>
          <label
            className="a-label a-label__heading body-copy">
            From
          </label>
          <div
            className="m-btn-inside-input">
            <input
              aria-describedby="input-error_message-from"
              className="a-text-input"
              max={null}
              min={2014-01-01T12:00:00.000Z}
              onChange={[Function]}
              placeholder="mm/dd/yyyy"
              size="15"
              value="" />
          </div>
        </li>
        <li>
          <label
            className="a-label a-label__heading body-copy">
            Through
          </label>
          <div
            className="m-btn-inside-input">
            <input
              aria-describedby="input-error_message-through"
              className="a-text-input"
              max={null}
              min={2014-01-01T12:00:00.000Z}
              onChange={[Function]}
              placeholder="mm/dd/yyyy"
              size="15"
              value="" />
          </div>
        </li>
      </ul>
    </div>
  </section>
  <hr />
  <section
    className="o-expandable aggregation">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        Matched Company Name
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
          onClick={[Function]}>
          Hide
          <span
            className="cf-icon cf-icon-minus-round" />
        </button>
      </span>
    </div>
    <p>
      The company name as it appears in our complaint system, which may be different than the name the consumer provided in their complaint
    </p>
    <section
      className="typeahead "
      onBlur={[Function]}
      onFocus={[Function]}>
      <div
        className="m-btn-inside-input">
        <input
          autoComplete="off"
          className="a-text-input"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Enter company name"
          type="text"
          value="" />
      </div>
    </section>
    <ul />
  </section>
  <hr />
  <section
    className="o-expandable aggregation">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        Product / sub-product
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
          onClick={[Function]}>
          Hide
          <span
            className="cf-icon cf-icon-minus-round" />
        </button>
      </span>
    </div>
    <p>
      The type of product and sub-product the consumer identified in the complaint
    </p>
    <p>
      <a
        className="a-btn__link"
        href="http://files.consumerfinance.gov/f/documents/201704_cfpb_Summary_of_Product_and_Sub-product_Changes.pdf"
        target="_blank">
        Recent changes to products and sub-products
      </a>
    </p>
    <div>
      <ul />
    </div>
  </section>
  <hr />
  <section
    className="o-expandable aggregation">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        Issue / sub-issue
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
          onClick={[Function]}>
          Hide
          <span
            className="cf-icon cf-icon-minus-round" />
        </button>
      </span>
    </div>
    <p>
      The type of issue and sub-issue the consumer identified in the complaint
    </p>
    <section
      className="typeahead "
      onBlur={[Function]}
      onFocus={[Function]}>
      <div
        className="m-btn-inside-input">
        <input
          autoComplete="off"
          className="a-text-input"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Enter name of issue"
          type="text"
          value="" />
      </div>
    </section>
    <div>
      <ul />
    </div>
  </section>
  <hr />
  <section
    className="o-expandable aggregation">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        State
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
          onClick={[Function]}>
          Hide
          <span
            className="cf-icon cf-icon-minus-round" />
        </button>
      </span>
    </div>
    <p>
      The state of the mailing address provided by the consumer
    </p>
    <section
      className="typeahead "
      onBlur={[Function]}
      onFocus={[Function]}>
      <div
        className="m-btn-inside-input">
        <input
          autoComplete="off"
          className="a-text-input"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Enter state name or abbreviation"
          type="text"
          value="" />
      </div>
    </section>
    <ul />
  </section>
  <hr />
  <section
    className="o-expandable aggregation">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        Zip Code
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
          onClick={[Function]}>
          Hide
          <span
            className="cf-icon cf-icon-minus-round" />
        </button>
      </span>
    </div>
    <p>
      The mailing ZIP code provided by the consumer
    </p>
    <section
      className="typeahead "
      onBlur={[Function]}
      onFocus={[Function]}>
      <div
        className="m-btn-inside-input">
        <input
          autoComplete="off"
          className="a-text-input"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Enter first three digits of zip code"
          type="text"
          value="" />
      </div>
    </section>
    <ul />
  </section>
  <hr />
  <section
    className="o-expandable aggregation">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        Did company provide a timely response?
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
          onClick={[Function]}>
          Show
          <span
            className="cf-icon cf-icon-plus-round" />
        </button>
      </span>
    </div>
    <p />
  </section>
  <hr />
  <section
    className="o-expandable aggregation">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        Company Response
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
          onClick={[Function]}>
          Show
          <span
            className="cf-icon cf-icon-plus-round" />
        </button>
      </span>
    </div>
    <p>
      How the company responded to the complaint
    </p>
  </section>
  <hr />
  <section
    className="o-expandable aggregation">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        Company Public Response
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
          onClick={[Function]}>
          Show
          <span
            className="cf-icon cf-icon-plus-round" />
        </button>
      </span>
    </div>
    <p>
      The company\'s optional public-facing response to a consumer\'s complaint
    </p>
  </section>
  <hr />
  <section
    className="o-expandable aggregation date-filter">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        Date complaint sent to company
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
          onClick={[Function]}>
          Hide
          <span
            className="cf-icon cf-icon-minus-round" />
        </button>
      </span>
    </div>
    <p />
    <div>
      <ul
        className="date-inputs">
        <li>
          <label
            className="a-label a-label__heading body-copy">
            From
          </label>
          <div
            className="m-btn-inside-input">
            <input
              aria-describedby="input-error_message-from"
              className="a-text-input"
              max={null}
              min={2014-01-01T12:00:00.000Z}
              onChange={[Function]}
              placeholder="mm/dd/yyyy"
              size="15"
              value="" />
          </div>
        </li>
        <li>
          <label
            className="a-label a-label__heading body-copy">
            Through
          </label>
          <div
            className="m-btn-inside-input">
            <input
              aria-describedby="input-error_message-through"
              className="a-text-input"
              max={null}
              min={2014-01-01T12:00:00.000Z}
              onChange={[Function]}
              placeholder="mm/dd/yyyy"
              size="15"
              value="" />
          </div>
        </li>
      </ul>
    </div>
  </section>
  <hr />
  <section
    className="o-expandable aggregation">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        Consumer Consent
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
          onClick={[Function]}>
          Show
          <span
            className="cf-icon cf-icon-plus-round" />
        </button>
      </span>
    </div>
    <p>
      Whether a consumer opted to publish their compaint narrative
    </p>
  </section>
  <hr />
  <section
    className="o-expandable aggregation">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        How did the consumer submit the complaint to the CFPB?
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
          onClick={[Function]}>
          Show
          <span
            className="cf-icon cf-icon-plus-round" />
        </button>
      </span>
    </div>
    <p />
  </section>
  <hr />
  <section
    className="o-expandable aggregation">
    <div
      className="o-expandable_header o-expandable_target">
      <h5
        className="o-expandable_header-left o-expandable_label">
        Tags
      </h5>
      <span
        className="o-expandable_header-right o-expandable_link">
        <button
          className="a-btn a-btn__link o-expandable_cue o-expandable_cue-open"
          onClick={[Function]}>
          Show
          <span
            className="cf-icon cf-icon-plus-round" />
        </button>
      </span>
    </div>
    <p>
      Data that supports easier searching and sorting of complaints submitted by or on behalf of older Americans and/or servicemembers
    </p>
  </section>
</section>
`;
